zarr/tests/test_core.py

```diff
@@ -62,16 +64,59 @@
 # noinspection PyMethodMayBeStatic
 
 
-class TestArray(unittest.TestCase):
-
+class TestArray():
     version = 2
     root = ''
-    KVStoreClass = KVStore
+    path = ''
+    compressor = Zlib(level=1)
+    filters = None
+    dimension_separator: Literal["/", ".", None] = None
+    cache_metadata = True
+    cache_attrs = True
+    partial_decompress: bool = False
+    write_empty_chunks = True
+    read_only = False
+    storage_transformers: Tuple[Any, ...] = ()
+
+    def create_store(self) -> BaseStore:
+        return KVStore(dict())
+
+    # used by child classes
+    def create_chunk_store(self) -> Optional[BaseStore]:
+        return None
+
+    def create_storage_transformers(self, shape) -> Tuple[Any, ...]:
+        return ()
+
+    def create_array(self, shape: Union[int, Tuple[int, ...]], **kwargs):
+        store = self.create_store()
+        chunk_store = self.create_chunk_store()
+        # keyword arguments for array initialization
+        init_array_kwargs = {
+            "path": kwargs.pop("path", self.path),
+            "compressor": self.compressor,
```

Comment:
Why not 
```suggestion
            "compressor": kwargs.pop("compressor", self.compressor),
```
?

Just trying to understand how this class is intended to be used by developers...